docs/source/guides/define_by_run.rst

```diff
@@ -0,0 +1,23 @@
+Define-by-run
+-------------
+
+.. currentmodule:: chainer
+
+As mentioned on the top page, Chainer is a flexible framework for neural networks.
+One major goal is flexibility, so it must enable us to write complex architectures simply and intuitively.
+
+Most existing deep learning frameworks are based on the **"Define-and-Run"** scheme.
+That is, first a network is defined and fixed, and then the user periodically feeds it with mini-batches.
```

Comment:
Although somewhat clear from the context, how about _...mini-batches of training data._ .